app/services/AuditService.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+
+import java.time.Instant
+import java.util.UUID
+
+import audit.events.AuditEventConstants.buildTags
+import audit.events.TagSet.ALL_TAGS
+import audit.events.{AuditEventConstants, ContactDetailsAuditEventDetail, TagSet}
+import javax.inject.{Inject, Singleton}
+import models.CompanyContactDetails
+import play.api.libs.json.{JsObject, Json, Writes}
+import play.api.mvc.{AnyContent, Request}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions.auditHeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AuditServiceImpl @Inject()(val auditConnector: AuditConnector) extends AuditService {
+}
+
+
+abstract class RegistrationAuditEvent(auditType: String, detail: JsObject, tagSet: TagSet = ALL_TAGS)(implicit hc: HeaderCarrier, req: Request[AnyContent])
+  extends ExtendedDataEvent(
+    auditSource = "company-registration-frontend",
+    auditType = auditType,
+    detail = detail,
+    tags = buildTags(auditType, tagSet)
+  )
+
+object RegistrationAuditEventTest {
+
+}
```

Comment:
These can be removed, because we want to pass the models directly to the new `sendEvent[T]` - this method takes care of ensure that the correct Tags are added.